### ISSUE TITLE
Triton Provider: firewall_enabled flag is always false

### ISSUE THREAD

**Issue by sodre** [hashicorp/terraform#6109]
The Triton provider is not picking up the correct state of the firewall in a machine, it always reports it as false.

Steps to reproduce:
 - Get an account at Joyent's public cloud and configure it as a provider. You will also need Triton's CLI.
 - Create a machine and firewall rule, e.g. 
```
resource "triton_firewall_rule" "bastion-to-vms" {
  rule = "FROM tag role=bastion TO all vms ALLOW tcp PORT 22"
  enabled = true
}

resource "triton_machine" "bastion" {
  name = "bastion"
  package = "t4-standard-512M"
  image = "ffe82a0a-83d2-11e5-b5ac-f3e14f42f12d"
  tags = { 
    role = "bastion"
  } 
  firewall_enabled = true        
  count = 1
}
```
- Run `terraform plan` and `terraform apply`. Everything works fine.
- Run `terraform show | grep firewall_enabled`. It will show "false"
- Run Joyent's `triton inst get bastion | grep firewall_enabled`. It will show "true"

If we run `terraform plan` again, it will create a rule trying to switch the firewall back to enabled, and will recreate the firewall resources.

**Comment by sodre** [comment]
According to joyent/gosdc#23 this is an issue with the vendor API.

**Comment by jen20** [comment]
Should be fixed by #6119.
